homeassistant/components/sensorpush/data.py

```diff
@@ -0,0 +1,85 @@
+"""The SensorPush Bluetooth integration."""
+from __future__ import annotations
+
+from sensorpush_ble import (
+    ATTR_MANUFACTURER as SENSOR_MANUFACTURER,
+    ATTR_MODEL as SENSOR_MODEL,
+    ATTR_NAME as SENSOR_NAME,
+    SIGNAL_STRENGTH_KEY,
+    DeviceClass,
+    DeviceKey,
+    SensorDeviceInfo,
+    SensorUpdate,
+)
+
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntityDescription
+from homeassistant.const import ATTR_MANUFACTURER, ATTR_MODEL, ATTR_NAME
+from homeassistant.helpers.entity import DeviceInfo
+
+SENSOR_DEVICE_CLASS_TO_HASS = {
+    key: getattr(SensorDeviceClass, key.value)
+    for key in DeviceClass
+    if hasattr(SensorDeviceClass, key.value)
+}
+
+
+def _device_key_to_bluetooth_entity_key(
+    device_key: DeviceKey,
+) -> PassiveBluetoothEntityKey:
+    """Convert a device key to an entity key."""
+    return PassiveBluetoothEntityKey(device_key.key, device_key.device_id)
+
+
+def _sensor_device_class_to_hass(
+    sensor_device_class: DeviceClass | None,
+) -> SensorDeviceClass | None:
+    """Convert a sensor device class to a sensor device class."""
+    if sensor_device_class is None:
+        return None
+    return SENSOR_DEVICE_CLASS_TO_HASS.get(sensor_device_class)
+
+
+def _sensor_device_info_to_hass(
+    sensor_device_info: SensorDeviceInfo,
+) -> DeviceInfo:
+    """Convert a sensor device info to a sensor device info."""
+    hass_device_info = DeviceInfo({})
+    if sensor_device_info.get(SENSOR_NAME) is not None:
+        hass_device_info[ATTR_NAME] = sensor_device_info[SENSOR_NAME]
+    if sensor_device_info.get(SENSOR_MANUFACTURER) is not None:
+        hass_device_info[ATTR_MANUFACTURER] = sensor_device_info[SENSOR_MANUFACTURER]
+    if sensor_device_info.get(SENSOR_MODEL) is not None:
+        hass_device_info[ATTR_MODEL] = sensor_device_info[SENSOR_MODEL]
+    return hass_device_info
+
+
+def sensor_update_to_bluetooth_data_update(
+    sensor_update: SensorUpdate,
+) -> PassiveBluetoothDataUpdate:
+    """Convert a sensor update to a bluetooth data update."""
+    return PassiveBluetoothDataUpdate(
+        devices={
+            device_id: _sensor_device_info_to_hass(device_info)
+            for device_id, device_info in sensor_update.devices.items()
+        },
+        entity_descriptions={
+            _device_key_to_bluetooth_entity_key(device_key): SensorEntityDescription(
+                key=f"{device_key.key}_{device_key.device_id}",
+                name=sensor_description.name,
+                device_class=_sensor_device_class_to_hass(
+                    sensor_description.device_class
+                ),
+                native_unit_of_measurement=sensor_description.native_unit_of_measurement,
```

Comment:
The unit of measurement should be a Home Assistant constant. Another reason why we should try to declare the entity descriptions.

Reply:
I refactored it.  Let me know what you think.  If it looks good, I'll go ahead and do the other 2 I have in the queue after this.

Reply:
Perfect! :tada: